data-reconciliation-app/docs/data-reconciliation.md

```diff
@@ -62,6 +62,127 @@ The reconciliation application will consist of three main services.
 
 ![reconciliation-sample diagram](reconciliation-sample.png)
 
+## Pseudo code
+
+Sample code in Typescript to demonstrate how to generate a voting-based data reconciliation report
+
+```typescript
+// generate a voting based data reconciliation report
+// memberId: the member who is requesting the report
+public getVotingSummaries(memberId: string) {
+
+    const summaries = new Map<string, VoteSummary>();
+
+    // iterate the key-value store to generate a data reconciliation summary record
+    // key-value store sample data record :
+    // key: Unique identifier, value : { attributes: {name: "XYZ", type: "string"}, votes: {"Member 1": A ,"Member 2": A ,"Member 3": A ,"Member 4": D  }}
+    // or
+    // key: Unique identifier, value : { attributes: {name: "XYZ", type: "number"}, votes: {"Member 1": 10 ,"Member 2": 20 ,"Member 3": 30 ,"Member 4": 40 }}
+    this.kvStore.forEach((val, key) => {
+        const type = val.attributes.type;
+        if(type == "string"){
+            summaries.set(key, getStringVotingSummary(memberId, key));
+        } else if(type == "number"){
+            summaries.set(key, getNumericVotingSummary(memberId, key));
+        }
+    });
+
+    return summaries;
+  }
+
+```
+
+### String Data Reconciliation
+
+```typescript
+// generate voting summary for string data type votes
+// memberId: the member who is requesting the data reconciliation summary report
+// recordId: the record identifier for the data to be reconciled
+function getStringVotingSummary(
+  memberId: string,
+  recordId: string
+): StringVoteSummary {
+  // record = { attributes: {name: "XYZ", type: "string"}, votes: {"Member 1": A ,"Member 2": A ,"Member 3": A ,"Member 4": D  }}
```

Comment:
We will likely need to change this record object structure a little bit...but regardless this logic is very helpful!